app/cdap/components/Administration/TetheringTabContent/NewTetheringRequest/NamespacesTable.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+import { Grid, GridHeader, GridBody, GridRow, GridCell, StyledCheckbox } from '../shared.styles';
+import { INamespace } from '../types';
+import {
+  OMNI_NS_COLUMN_TEMPLATE,
+  OMNI_NS_COLUMN_TEMPLATE_SCROLLABLE,
+  OMNI_NS_TABLE_HEADERS,
+} from './constants';
+
+interface INameSpacesTableProps {
+  tableData: INamespace[];
+  selectedNamespaces: string[];
+  broadcastChange: (ns: string) => void;
+}
+
+const StyledGrid = styled(Grid)`
+  padding: 0;
+  margin-bottom: 10px;
+`;
+
+const StyledGridBody = styled(GridBody)`
+  max-height: 150px;
+  overflow: auto;
+`;
+
+const NameSpacesTable = ({
+  tableData,
+  selectedNamespaces,
+  broadcastChange,
+}: INameSpacesTableProps) => {
+  const renderTableHeader = () => {
```

Comment:
could this and the function below be pulled out of this component to prevent declaring it each render?

Reply:
Like this? 
```javascript
const renderTableHeader = (tableData: INamespace[]) => {
  const columnTemplate =
    tableData.length > 5 ? OMNI_NS_COLUMN_TEMPLATE_SCROLLABLE : OMNI_NS_COLUMN_TEMPLATE;
  return (
    <GridHeader>
      <GridRow columnTemplate={columnTemplate}>
        {OMNI_NS_TABLE_HEADERS.map((header, i) => {
          return <GridCell key={i}>{header.label}</GridCell>;
        })}
      </GridRow>
    </GridHeader>
  );
};

const renderTableBody = (
  data: INamespace[],
  renderRowFn: (req: INamespace, idx: number) => React.ReactNode
) => <StyledGridBody>{data.map((req, idx) => renderRowFn(req, idx))}</StyledGridBody>;

const NameSpacesTable = ({...})
```

Reply:
Yeah

Reply:
Addressed in [e9c26f3](https://github.com/cdapio/cdap-ui/pull/342/commits/e9c26f34fc0246a1283492c17710ef7cb2890a84)